src/lib/utils/scales-utils.js

```diff
@@ -615,3 +624,24 @@ export function getMissingScaleProps(props, data, attributes) {
   });
   return result;
 }
+
+/**
+ * Return a d3 scale that returns the literal value that was given to it
+ * @returns {function} literal scale.
+ */
+export function literalScale() {
+  function scale(d) {
+    return d;
+  }
+
+  function response() {
+    return scale;
+  }
+
+  scale.domain = response;
+  scale.range = response;
+  scale.unknown = response;
+  scale.copy = response;
+
+  return scale;
```

Comment:
Since scale is immutable, I'd suggest to move it outside of the `literalScale` function